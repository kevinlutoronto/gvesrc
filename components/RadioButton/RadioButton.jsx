// @flow
import React from 'react';
import cx from 'classnames';
import Icon from '../Icon/Icon';
import { getDataAttrs } from '../../dataUtils';
import type { Data } from '../../types';
import style from './style.scss';

type Props = {|
  id?: string,
  data?: Data,
  checked?: boolean,
  description?: ?string,
  disabled?: ?boolean,
  fancy?: ?boolean,
  fieldValue?: string,
  icon?: ?string,
  isLarge?: ?boolean,
  name: string,
  onChange: (event: SyntheticEvent<HTMLInputElement>) => mixed,
  text: React$Node,
  value?: string,
|};

class RadioButton extends React.Component<Props> {
  static defaultProps = {
    name: 'radioButton',
    text: 'Option',
  };

  value = (): mixed => {
    const { checked, value, fieldValue } = this.props;
    return checked || (fieldValue && value === this.props.fieldValue);
  };

  render() {
    const {
      data,
      description,
      disabled,
      fancy,
      fieldValue,
      icon,
      isLarge,
      name,
      onChange,
      text,
      id,
    } = this.props;
    const wrapperClasses = cx(style.wrapper, {
      [style.disabled]: disabled,
      [style.fancy]: fancy,
      [style.fancyChecked]: fancy && this.value(),
      [style.isLarge]: isLarge,
    });
    const inlineClass = cx({ [style.inline]: fancy });
    const fauxCircle = fancy ? null : <div className={style.faux} />;
    const fancyIcon =
      fancy && icon ? (
        <div className={style.fancyIcon}>
          <Icon icon={icon} />
        </div>
      ) : null;
    return (
      <div {...getDataAttrs(data)} className={inlineClass}>
        <label htmlFor={id} className={wrapperClasses}>
          <input
            className={cx(style.original, {
              [style.disabled]: disabled,
            })}
            checked={this.value()}
            disabled={disabled}
            id={id}
            name={name}
            onChange={onChange}
            type="radio"
            value={fieldValue}
          />
          {fauxCircle}
          {fancyIcon}
          <div
            className={cx(style.text, {
              [style.disabled]: disabled,
            })}
          >
            {text}
          </div>
        </label>
        {description && <span className={style.description}>{description}</span>}
      </div>
    );
  }
}

RadioButton.displayName = 'Plasma@RadioButton';

export default RadioButton;
