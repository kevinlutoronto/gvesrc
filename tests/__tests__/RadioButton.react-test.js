/* eslint-disable */

import React from 'react';
import RadioButton from '../../components/RadioButton/RadioButton';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

describe('RadioButton Component', () => {
  test('render', () => {
    const component = renderer.create(<RadioButton />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    tree.props.checked = true;
    expect(tree).toMatchSnapshot();
  });

  test('description', () => {
    const component = shallow(<RadioButton description="wow such description" />);
    expect(component.find('.description')).toHaveLength(1);
  });
});
