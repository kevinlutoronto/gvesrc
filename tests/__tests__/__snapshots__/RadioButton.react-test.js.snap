// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton Component render 1`] = `
<div
  className=""
>
  <label
    className="wrapper"
    htmlFor={undefined}
  >
    <input
      checked={undefined}
      className="original"
      disabled={undefined}
      id={undefined}
      name="radioButton"
      onChange={undefined}
      type="radio"
      value={undefined}
    />
    <div
      className="faux"
    />
    <div
      className="text"
    >
      Option
    </div>
  </label>
</div>
`;

exports[`RadioButton Component render 2`] = `
<div
  checked={true}
  className=""
>
  <label
    className="wrapper"
    htmlFor={undefined}
  >
    <input
      checked={undefined}
      className="original"
      disabled={undefined}
      id={undefined}
      name="radioButton"
      onChange={undefined}
      type="radio"
      value={undefined}
    />
    <div
      className="faux"
    />
    <div
      className="text"
    >
      Option
    </div>
  </label>
</div>
`;
